src/runtime_src/xrt/device/device.h

```diff
@@ -810,6 +810,12 @@ class device
     return m_hal->stopTrace(type);
   }
 
+  hal::operations_result<size_t>
```

Comment:
This is not an appropriate C++ interface.  Please use const appropriately and use std::array for wrapping fixed length char* arrays.  But I think this function should return a std::string with the path, not size_t.  So signature should be hal_operations<std::string>
getSysfsPath(const std::string& subdev, const std::stirng& entry);

Reply:
changed to use `hal::operations_result<std::string>` in the new push